packages/hooks/use-archive-infinite-scroll/index.tsx

```diff
@@ -0,0 +1,190 @@
+  // browser state, we use those. If not, we start with the initial link. The
+  // same for the limit.
+  const links: string[] = infiniteScroll?.links
+    ? [...infiniteScroll?.links]
+    : [state.router.link];
+  const limit: number = infiniteScroll?.limit || options.limit;
+
+  // Initialize/update browser state.
+  useEffect(() => {
+    if (!options.active) return;
+
+    actions.router.updateState({
+      ...state.router.state,
+      infiniteScroll: {
+        links,
+        limit,
+        ...infiniteScroll,
+      },
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [options.active]);
+
+  // Aliases to needed state.
+  const last = state.source.get(links[links.length - 1]);
+  const next =
+    isArchive(last) && last.next ? state.source.get(last.next) : null;
+
+  // Infinite scroll booleans.
+  const hasReachedLimit = !!limit && links.length >= limit;
+  const thereIsNext = isArchive(last) && last.next;
+  const isFetching = last.isFetching || !!next?.isFetching;
+  const isLastError = isError(last);
+  const isLimit = hasReachedLimit && thereIsNext && !isFetching;
+
+  /**
+   * Function to fetch the next item disregarding the limit.
+   */
+  const fetchNext = async () => {
+    if (
+      // Don't fetch if this hook is not active anymore.
+      !options.active ||
+      // Don't fetch if there is no next link and the last one was not an error.
+      // This is because we use the same function to refetch the last link if
+      // there was an error.
+      (!thereIsNext && !isLastError) ||
+      // Don't fetch if the next item is already included in the links. This
+      // means it has been already fetched or it won't be there.
+      (isArchive(last) && links.includes(last.next))
+    )
+      return;
+
+    if (isLastError) {
+      // If the last link returned an error, we use this function to repeat the
+      // fetch.
+      await actions.source.fetch(links[links.length - 1], { force: true });
+    } else if (isArchive(last)) {
+      links.push(last.next);
+
+      actions.router.updateState({
+        ...state.router.state,
+        infiniteScroll: {
+          ...infiniteScroll,
+          links,
+        },
+      });
+
+      const next = state.source.get(last.next);
+
+      if (!next.isReady && !next.isFetching) {
+        await actions.source.fetch(last.next);
+      }
+    }
+  };
+
+  // Map every link to its DIY object.
+  const pages = links.map((link) => ({
+    key: link,
+    link: link,
+    isLast:
+      (link === last.link && !!last.isReady && !isError(last)) ||
+      (link === links[links.length - 2] && !last.isReady) ||
+      (link === links[links.length - 2] && !!last.isReady && !!isError(last)),
+    Wrapper: options.active
+      ? MemoizedWrapper(link, {
+          fetchInViewOptions: options.fetchInViewOptions,
+          routeInViewOptions: options.routeInViewOptions,
+        })
+      : ({ children }) => <>{children}</>,
```

Comment:
In this hook, if it is not active we don't use the `Wrapper`, which I think makes sense. But in the other one, we use it always, even when it is active.

https://github.com/frontity/frontity/blob/29730058958ab880c2563dce3f135eeb76d72b99/packages/hooks/use-post-type-infinite-scroll/index.tsx#L329-L332

Is there a reason for that?

Reply:
Well, I'm pretty sure I changed that for the `useArchiveInfiniteScroll` because the next page was being fetched even though the hook was disabled, as the `<Wrapper>` is the one in charge of doing that (the wrapper is not disabled).

The same doesn't happened with the `usePostTypeInfiniteScroll` because the next page is fetched when you reach the penultimate post. But, it could happen if you are in the penultimate post and then you disable the hook... :thinking:

I'm going to add a test for that and fix it.

Reply:
Perfect, thank you! It's not a bid deal but I think it will be more solid that way 🙂

Reply:
It turns out I was already doing that test. :joy: I added a missing test anyway to check that the next post wasn't rendered or fetched when the hook is deactivated, and It worked.

Reply:
Perfect, thank you!